peer/x/peerheap/heap.go

```diff
@@ -0,0 +1,105 @@
+package peerheap
+
+import (
+	"container/heap"
+	"fmt"
+
+	"go.uber.org/yarpc/api/peer"
+)
+
+type peerHeap struct {
+	transport peer.Transport
+	peers     []*peerScore
+
+	// next is an incrementing counter for every push, which is compared when
+	// scores are equal. This ends up implementing round-robin when scores are
+	// equal.
+	next int
```

Comment:
We may need to revisit overflow. We can probably get around the problem by normalizing "last" values around "next" when computing scores.